marklogic-client-api/src/test/java/com/marklogic/client/test/dataservices/BulkInputCallerTest.java

```diff
@@ -0,0 +1,118 @@
+package com.marklogic.client.test.dataservices;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.client.dataservices.ExecEndpoint;
+import com.marklogic.client.dataservices.InputEndpoint;
+import com.marklogic.client.document.JSONDocumentManager;
+import com.marklogic.client.io.FileHandle;
+import com.marklogic.client.io.JacksonHandle;
+import org.hamcrest.core.StringContains;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.internal.matchers.ThrowableMessageMatcher;
+import org.junit.rules.ExpectedException;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.*;
+
+public class BulkInputCallerTest {
+
+    static ObjectNode apiObj;
+    static String apiName = "bulkInputCallerImpl.api";
+    static String scriptPath;
+    static String apiPath;
+    static int workMax = 5;
+    static int startValue = 1;
+    static JSONDocumentManager docMgr;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
```

Comment:
Would it be more robust to initialize expectedException to ExpectedException.none() within bulkInputEndpointTest()?

If someone added a test that expects an exception before bulkInputEndpointTest(), would that cause bulkInputEndpointTest() to fail?

Reply:
Hi Erik. I guess it should be okay. I tried adding a unit test to throw null pointer before bulkInputEndpointTest() and it ran successfully. Also, if we move the initialization of expectedException to bulkInputEndpointTest() we will need to initialize it again within bulkInputEndpointInterruptTest().